src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,322 @@
+        if toolchain_override::Channel::from_str(&input_channel_name).is_err() {
+            bail!(
+                "Invalid channel '{}', expected one of {}.",
+                input_channel_name,
+                VALID_CHANNEL_STR,
+            );
+        } else {
+            export_channel = input_channel_name;
+        }
+    }
+
+    let toolchain_override = ToolchainOverride {
+        cfg: OverrideCfg::new(
+            ToolchainCfg {
+                // here shouldn't be err as we has checked above
+                channel: toolchain_override::Channel::from_str(&export_channel)?,
+            },
+            Some(version_map),
+        ),
+        path: toolchain_info_path.clone(),
+    };
+    let document = toolchain_override.to_toml();
+    if std::fs::write(toolchain_override.path, document.to_string()).is_err() {
+        bail!("Failed to write {}", toolchain_info_path.display());
+    }
+
+    info!(
+        "Exported toolchain \"{}\" to file: \"{}\"",
+        toolchain_name,
+        toolchain_info_path.display()
+    );
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        channel, path::settings_file, toolchain::DistToolchainName,
+        toolchain_override::ToolchainOverride,
+    };
+    use serial_test::serial;
+    use std::{fs, io::Write};
+
+    use super::*;
+
+    // simulate input
+    const INPUT_NOP: &[u8; 1] = b"\n";
+    const INPUT_YES: &[u8; 2] = b"y\r";
+    const INPUT_NO: &[u8; 4] = b"n \r\n";
+    const INPUT_INVALID_CHANNEL: &[u8; 11] = b"my-channel\n";
+    const INVALID_CHANNEL: &str = "my-channel";
+
+    fn remove_toolchain_info() {
+        if let Some(fuel_toolchain_toml_file) = get_fuel_toolchain_toml() {
+            if fuel_toolchain_toml_file.exists() {
+                fs::remove_file(fuel_toolchain_toml_file).unwrap();
+            }
+        }
+    }
+    fn create_toolchain_info() {
+        if let Some(fuel_toolchain_toml_file) = get_fuel_toolchain_toml() {
+            if !fuel_toolchain_toml_file.exists() {
+                fs::create_dir_all(fuel_toolchain_toml_file.parent().unwrap()).unwrap();
+                fs::File::create(&fuel_toolchain_toml_file).unwrap();
+            }
+        } else {
+            fs::File::create(FUEL_TOOLCHAIN_TOML_FILE).unwrap();
+        }
+    }
+    // mock setting.toml
+    fn create_toolchain_settings_file() {
+        let setting_file_path = settings_file();
+        if !setting_file_path.exists() {
+            fs::create_dir_all(setting_file_path.parent().unwrap()).unwrap();
+            let mut file = fs::File::create(&setting_file_path).unwrap();
+            file.write_all(b"default_toolchain = \"latest\"").unwrap();
+        }
+    }
+    fn assert_channel_name(expected: &String) {
+        let toolchain_override =
+            ToolchainOverride::from_project_root().expect("toolchain override");
+        let _actual = toolchain_override.cfg.toolchain.channel.to_string();
+        assert!(matches!(expected, _actual));
```

Comment:
```suggestion
        let actual = toolchain_override.cfg.toolchain.channel.to_string();
        assert!(matches!(expected, actual));
```

Reply:
reported warning by rustc lint 
> if this is intentional, prefix it with an underscore: `_actual`rustc(unused_variables)

Reply:
That's odd, maybe the compiler doesn't recognize the usage inside the `matches` macro. It would be better to use `assert_eq` here anyway.

Reply:
> That's odd, maybe the compiler doesn't recognize the usage inside the `matches` macro. It would be better to use `assert_eq` here anyway.

@sdankel Hi sir, suggestion changes addressed!